packages/compass-components/src/components/bson-value.tsx

```diff
@@ -0,0 +1,288 @@
+  | { type: 'Decimal128'; value: Decimal128 };
+
+type ValueProps =
+  | BSONValueProps
+  | { type: 'Date'; value: Date; tz?: string }
+  | { type: 'String'; value: string }
+  | { type: 'Undefined'; value: undefined }
+  | { type: 'Null'; value: null }
+  | { type: 'Boolean'; value: boolean };
+
+function truncate(str: string, length = 70): string {
+  const truncated = str.slice(0, length);
+  return length < str.length ? `${truncated}…` : str;
+}
+
+type ValueTypes = ValueProps['type'];
+
+type PropsByValueType<V extends ValueTypes> = Omit<
+  Extract<ValueProps, { type: V }>,
+  'type'
+>;
+
+function getClassName(
+  type: ValueTypes | Lowercase<ValueTypes> | string
+): string {
+  return `element-value element-value-is-${type.toLowerCase()}`;
+}
+
+const nonSelectable = css({
+  userSelect: 'none',
+});
+
+export const ObjectIdValue: React.FunctionComponent<
+  PropsByValueType<'ObjectId'>
+> = ({ value }) => {
+  const stringifiedValue = useMemo(() => {
+    return String(value);
+  }, [value]);
+
+  return (
+    <div className={getClassName('objectid')} title={stringifiedValue}>
+      <span className={nonSelectable}>ObjectId(&apos;</span>
+      {stringifiedValue}
+      <span className={nonSelectable}>&apos;)</span>
+    </div>
+  );
+};
+
+export const BinaryValue: React.FunctionComponent<PropsByValueType<'Binary'>> =
+  ({ value }) => {
+    const { stringifiedValue, title } = useMemo(() => {
+      if (value.sub_type === Binary.SUBTYPE_ENCRYPTED) {
+        return {
+          stringifiedValue: '*********',
+          title: 'Encrypted',
+        };
+      }
+      if (value.sub_type === Binary.SUBTYPE_UUID) {
+        return { stringifiedValue: `UUID('${value.toUUID().toHexString()}')` };
+      }
+      return {
+        stringifiedValue: `Binary('${truncate(
+          value.toString('base64'),
+          100
+        )}', ${value.sub_type})`,
+      };
+    }, [value]);
+
+    return (
+      <div className={getClassName('binary')} title={title ?? stringifiedValue}>
+        {stringifiedValue}
+      </div>
+    );
+  };
+
+export const CodeValue: React.FunctionComponent<PropsByValueType<'Code'>> = ({
+  value,
+}) => {
+  const stringifiedValue = useMemo(() => {
+    return `Code('${String(value.code)}'${
+      value.scope ? `, ${JSON.stringify(value.scope)}` : ''
+    })`;
+  }, [value.code, value.scope]);
+
+  return (
+    <div className={getClassName('code')} title={stringifiedValue}>
+      {stringifiedValue}
+    </div>
+  );
+};
+
+export const DateValue: React.FunctionComponent<PropsByValueType<'Date'>> = ({
+  value,
+  // NB: Compass (and dependants like Cloud) currently only support showing
+  // timezone in UTC, to avoid all the complexity of formatting with timezone
+  // included we ignore this logic for now, but we probably want to allow users
+  // to select timezone and display the output based on their settings. We can
+  // add support for this later when we really need / want to
+  // tz,
```

Comment:
I didn't want to add deprecated moment package as a dependency for compass-components and as this is not used at all in Compass (or in cloud / charts) and after checking with Max I decided to just disable this prop for now. When the time comes for us to add timezone support we can decide how exactly we will do this (it would probably make sense to move all our date helpers to one place at that point)

Reply:
Makes sense to me, is great that we are cleaning up weird deps, we probably don't need this comment either

Reply:
Will clean it up, I guess it's indeed more an explainer for the PR rather than something for the code